java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java

```diff
@@ -2107,14 +2113,17 @@ private ExpressionTree applySelectors(ExpressionTree primary, Optional<List<Expr
       for (ExpressionTree selector : selectors.get()) {
         if (selector.is(Kind.IDENTIFIER)) {
           IdentifierTreeImpl identifier = (IdentifierTreeImpl) selector;
-          result = new MemberSelectExpressionTreeImpl(result, identifier,
-            (AstNode) result, identifier);
+          // TODO The dot should have already been correctly handled
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)